File: Go/Encode-And-Decode-Strings.go
package main

import (
	"fmt"
	"strconv"
)

func encode(strs []string) string {
	result := ""
	for _, str := range strs {
		result += fmt.Sprint(len(str), "#", str)
	}
	return result
}

func decode(str string) []string {
	decodedWords := []string{}
	i := 0
	for i < len(str) {
		j := i
		for string(str[j]) != "#" {
			j++
		}
		len, _ := strconv.Atoi(str[i:j])
		word := str[j+1 : j+1+len]
		decodedWords = append(decodedWords, word)
		i = j + 1 + len
	}
	return decodedWords
}

File: Javascript/Encode-And-Decode-Strings.js
/**
 *
 * @param {string[]} strs
 * @returns {string}
 */
const encode = (strs) => {
  return strs.map((str) => `${str.length}#${str}`).join("");
};

/**
 *
 * @param {string} str
 * @returns {string[]}
 */
const decode = (str) => {
  let decodedWords = [];
  let i = 0;

  while (i < str.length) {
    let j = i;
    while (str[j] !== "#") {
      j++;
    }
    let len = parseInt(str.slice(i, j), 10);
    decodedWords.push(str.slice(j + 1, j + 1 + len));
    i = j + 1 + len;
  }
  return decodedWords;
};
